src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePumpSettings.cs

```diff
@@ -81,8 +83,17 @@ private string GetConnectionStringFromConfiguration()
 
         private async Task<string> GetConnectionStringFromSecretAsync()
         {
-            var secretProvider = _serviceProvider.GetRequiredService<ISecretProvider>();
-            return await _getConnectionStringFromSecretFunc(secretProvider);
+            ISecretProvider userDefinedSecretProvider = _serviceProvider.GetService<ICachedSecretProvider>()
+                                                        ?? _serviceProvider.GetService<ISecretProvider>();
+
+            if (userDefinedSecretProvider == null)
+            {
+                throw new KeyNotFoundException(
```

Comment:
😄 I recognize my own code. Thanks for the trust.

Reply:
Thank YOU!